src/backend/Functions/Analyze.cs

```diff
@@ -1,38 +1,51 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.WebJobs;
-using Microsoft.Azure.WebJobs.Extensions.Http;
-using Microsoft.Azure.WebJobs.Host;
-using Microsoft.Fx.Portability;
-using Microsoft.Fx.Portability.ObjectModel;
 using System;
+using System.IO;
+using System.IO.Compression;
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Fx.Portability;
+using Microsoft.Fx.Portability.ObjectModel;
+using Newtonsoft.Json;
+using WorkflowManagement;
 
 namespace Functions
 {
     public static class Analyze
     {
+        //Allows ActionFactory to be "injected" as needed, such as with a mock action factory when the function is called through tests
+        public static Func<WorkflowManager> GetWorkflowManager { get; set; } = () => new WorkflowManager();
+
         [FunctionName("analyze")]
         public static async Task<HttpResponseMessage> Run(
             [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req,
-            TraceWriter log)
+            [Queue("apiportworkflowqueue")]ICollector<WorkflowQueueMessage> workflowMessageQueue, 
+            ILogger log)
         {
             var analyzeRequest = await DeserializeRequest(req.Content);
             if (analyzeRequest == null)
             {
-                log.Error("invalid request");
+                log.LogError("invalid request");
                 return req.CreateResponse(HttpStatusCode.BadRequest);
             }
 
             var submissionId = Guid.NewGuid().ToString();
-            log.Info($"created submission id {submissionId}");
+            log.LogInformation($"created submission id {submissionId}");
```

Comment:
If this is structured logging, then you need to provide a an argument at the end instead of using interpolated strings. ie `log.LogInformation("Created submission id {SubmissionId}", submissionId);`. That way, things such as the submissionId can be correlated later. See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?tabs=aspnetcore2x#log-message-template for more info